### PR TITLE
Terminate subprocesses when main proces is terminated.

### LINKED ISSUE #48390: SIGTERM -- Return error when salt-minion service/process receives a SIGTERM -- Feature Request
### Description of Issue/Question

Currently, when salt-minion service/process receives a SIGTERM, it just exits. 
It leaves all the running jobs in orphan state. 

**Feature-Request**: Please return some kind of errors for those jobs before stopping the minion.

### PR DESCRIPTION
**Pull request by DmitryKuzmenko**
### What does this PR do?
When minion's main process receives the SIGTERM or SIGINT it doesn't shut down the worker processes executing jobs.
While if minion is stopped by "Ctrl-C" from terminal it shuts down all the processes.
This PR removes this inconsistency.

Also I've fixed the `SubprocessList.cleanup` method where iteration was broken.
Also fixed Lint failure: `third party module 'psutil' is not surrounded by try/except` that appeared from pre-commit hook.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/48390

### Previous Behavior
Run
```bash
for i in $(seq 3); do salt '*' test.sleep 100 --async; done
```
This will run 3 async jobs on the salt minion. Now send the SIGTERM to the only main process:
```bash
pkill -f salt-minion$
```
Result: the minion reports about stopping the logging queue and the main minion process. Then it hangs for 100 seconds until the last job is done. Then exits.
Re-run the same but run minion in foreground and hit `Ctrl-C`. Minion exits immediately.

### New Behavior
Result: minion exits immediately terminating the jobs.

### Merge requirements satisfied?
- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/latest/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by bryceml** [comment]
re-run all full

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
re-run all full

**Comment by DmitryKuzmenko** [comment]
The tests are broken by `filename_map.yml`. Fix is here: #57004

**Comment by DmitryKuzmenko** [comment]
It looks tests timeout issue could be related. I'm checking it.

**Review by s0undt3ch — DISMISSED** [review]

File: tests/unit/utils/test_process.py [hunk 1]
```diff
@@ -535,16 +531,42 @@ def test_process_list_cleanup(self):
...
+        assert proc2 in plist.processes
         self.wait_for_proc(proc)
+        self.wait_for_proc(proc2)
         assert not proc.is_alive()
+        assert not proc2.is_alive()
         plist.cleanup()
         assert proc not in plist.processes
+        assert proc2 not in plist.processes
+
+    @skipIf(True, "SLOWTEST skip")
```
Comment (s0undt3ch) [inline tests/unit/utils/test_process.py:553]: Let's add instead the slow test decorator?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/unit/utils/test_process.py (hunk 1 above)
Comment (DmitryKuzmenko) [reply tests/unit/utils/test_process.py:553]: Sure! Thank you @s0undt3ch !

Comment (DmitryKuzmenko) [reply tests/unit/utils/test_process.py]: I wonder how it could be missed. :shrug:. Fixed.

**Review by s0undt3ch — DISMISSED** [review]

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**